packages/table/src/TableBody.tsx

```diff
@@ -1,17 +1,57 @@
-import React from 'react';
-import { useTableContext } from './TableContext';
+import React, { useMemo, useCallback } from 'react';
+import { getDataComparisonFunction, useTableContext } from './TableContext';
 import { TableProps } from './Table';
 
 type TableBodyProps<Shape> = Pick<TableProps<Shape>, 'children'>;
 
+/**
+ * Hook that avoids re-rendering children when the data used to
+ * render them hasn't changed, even if their ordering has changed.
+ */
+function useRenderedChildren<Props>(
+  propList: Array<Props>,
```

Comment:
I'm confused by this variable name. Isn't this `data`? Why the rename?

Reply:
the variable name is generalized in case this hook is ever used in another context

Reply:
I struggle to imagine what other context could make use of this hook. For now can we rename to `data` (and `datum` inside the loop respectively) to make reading it in the context of Table clear? If we see a use for it elsewhere we can generalize it then.

Reply:
you could use this hook in any component that has a render prop whether that prop happens to be `children` or not. You can also use it anywhere a `map`, `forEach`, or `for (...)` is used to compute children or list of react elements. 

I don't really understand the line of reasoning that leads to over-specifying the naming. to me the whole point of extracting something out into a function rather than inlining it is to abstract away a high-level operation. the fact that a `prop` is a `datum` is just an implementation detail of `TableBody`. 
this
```ts
   const renderFunction = useCallback(datum => children({ datum }), [children]); 
   const rows = useRenderedChildren(data, renderFunction, compareFn); 
```
could have as easily also been
```ts
   const rows = useRenderedChildren(data.map(datum => ({datum})), children, compareFn); 
```

if it's not clear what the hook definition does in a vacuum, then I'd prefer feedback on how to make it more clear.

Reply:
Nothing's done in a vacuum, this is being written in the context of table so keeping consistent with those variables increases the overall readability of Table as a hole. 

We don't know how this will be applied generically (in a vacuum), however I do doubt a generic form would make use of a `compareFn`. That seems very Table specific itself.

I value readability highly, and the better we can serve that interest then the stronger the codebase is as a whole.

Reply:
>Nothing's done in a vacuum, this is being written in the context of table so keeping consistent with those variables increases the overall readability of Table as a hole.

but the calculus would be different if I had just defined the hook in the hooks package. then the emphasis of the review would be more so on improving its standalone clarity. I chose not to define it separately to reduce the scope of the PR change and reduce the burden of needing to unit test it more thoroughly.

> That seems very Table specific itself.

That's why I made `compareFn` optional, and thus ignorable, part of the API rather than `((a: Props, b: Props) => number) | undefined`. I could technically have split that part out but it would've been pretty unnatural since it either would have to reconstruct the mapping given the list of props and react nodes or have `useRenderedChildren` return the mapping.

Reply:
Seems like we're splitting hairs here? Also seems like the desire is to have this custom hook be reviewed independently is in tension with it needing to still be aware of Table specific needs, I.e. `compareFn`. 

I think what I found jarring when reading this was the introduction of this `propList` variable which is out of context for the rest of the Table component. It causes some mental strain to map what variables we're talking about when I made the jump into this function.

I don't think we're ready to pull this out to the hooks package, so my main ask is that we embrace its locality within Table and keep its variable naming consistent with the rest of Table.

Also - did moving to Map have any impact on performance? (Related - we should get some performance scaffolding in place at some point?).